### PR TITLE
Add lake_option to hydro_namelist, reservoirs to their own new namelist

### PR DESCRIPTION
**Pull request by rcabell**
TYPE: enhancement

KEYWORDS: lakes, reservoirs

SOURCE: NCAR

DESCRIPTION OF CHANGES: 

* Add `lake_option` (integer) to &hydro_nlist: 0 [lakes off], 1 [level pool], or 2 [passthrough], or 3 [reservoir DA]

  - turning lakes off (lake_option=0) will disable lakes even if route_lake_f is supplied, or outlake is turned on.

  - **Reservoir DA will not be used unless lake_option=3, even if all other required namelist options are present**

* Reservoir options have been moved from &hydro_nlist to &reservoir_nlist

   - This will make it easier to isolate / compose namelist files

   - If lake_option is not equal to 3, &reservoir_nlist won't be read, meaning it can be completely removed for applications that don't need it

TESTS CONDUCTED: In-progress, will convert from Draft and update here when complete.

### CODE REVIEW

**Comment by rcabell** [comment]
CI Tests failing due to namelist changes. Will update...

**Review by scrasmussen — APPROVED** [review]
Read through everything and it LGTM!

**Comment by rcabell** [comment]
I added a small fix to config.f90 to reset `lake_option` from 3 to 1 (level pool) after setting the `reservoir_type_specified` flag (which activates the various Reservoir DA schemes). I also added a message to `LEVELPOOL_PHYSICS()` to call `hydro_stop()` if an invalid lake_option is passed to it. 

This fix corrects the regression seen on Derecho when using Reservoir DA [not tested on GitHub CI].

**Review by scrasmussen — APPROVED** [review]
Those fixes and changes look good!

File: src/OrchestratorLayer/config.f90 [hunk 1]
```diff
@@ -223,258 +224,308 @@ subroutine rt_nlst_check(self)
...
    endif
 
-   if((self%channel_option .eq. 3) .and. (self%compound_channel)) then
+   if ((self%channel_option .eq. 3) .and. (self%compound_channel)) then
       call hydro_stop("Compound channel option not available for diffusive wave routing. ")
    end if
 
-   if(self%reservoir_type_specified) then
-      if(len(trim(self%reservoir_parameter_file)) .eq. 0) then
+   if ((self%lake_option .lt. 0) .and. (self%lake_option .gt. 3)) then
```
Comment (rcabell) [inline src/OrchestratorLayer/config.f90:478]: This should be .or. not .and.

File: src/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F90 [hunk 2]
```diff
@@ -201,7 +201,7 @@ subroutine hybrid_init(this, water_elevation,  &
             ! Initialize level pool reservoir
             call this%state%levelpool_ptr%init(water_elevation, lake_area, &
             weir_elevation, weir_coeffecient, weir_length, dam_length, orifice_elevation, &
-            orifice_coefficient, orifice_area, lake_max_water_elevation, lake_number)
+            orifice_coefficient, orifice_area, lake_max_water_elevation, lake_number, 1)
```
Comment (scrasmussen) [inline src/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F90:204]: I'm fine either way but do we want to make this 1 and the 1 in `module_rfc_forecasts.F90`:159 a named constant to better indicated what is being passed?

**Review by scrasmussen — APPROVED** [review]
Looks good to me! Did some testing and I things seemed to function properly

**Pull request merged**